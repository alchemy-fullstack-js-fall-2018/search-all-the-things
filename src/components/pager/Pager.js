import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';

import styles from './Pager.css';

export default class Pager extends Component {

  static propTypes = {
    currentPage: PropTypes.number.isRequired,
    totalPages: PropTypes.number.isRequired,
    updatePage: PropTypes.func.isRequired
  }

  render() {

    const { currentPage, totalPages, updatePage } = this.props;

    return (
      <div>

        {(currentPage > 1) &&
          <button
            className={styles.pageButton}
            onClick={() => updatePage(currentPage - 1)}
          >Previous
          </button>
        }

        <span>{currentPage} out of {totalPages} pages</span>

        {(currentPage < totalPages) &&
          <button
            className={styles.pageButton}
            onClick={() => updatePage(currentPage + 1)}
          >Next
          </button>
        }

      </div>
    );
  }
}
