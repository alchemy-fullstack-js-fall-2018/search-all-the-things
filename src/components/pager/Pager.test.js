import Pager from './Pager';

describe('Pager Component', () => {
  it('should be a pager', () => {
    const pager = new Pager({
      currentPage: 1,
      totalPages: 5,
      updatePage: () => { return null; }
    });
    expect(pager.render()).toMatchSnapshot();
  });
});
