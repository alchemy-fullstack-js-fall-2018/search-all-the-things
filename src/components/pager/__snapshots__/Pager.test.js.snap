// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pager Component should be a pager 1`] = `
<div>
  <span>
    1
     out of 
    5
     pages
  </span>
  <button
    className="pageButton"
    onClick={[Function]}
  >
    Next
  </button>
</div>
`;
